lighthouse/helpers/plates.py

```diff
@@ -263,6 +263,124 @@ def send_to_ss_heron_plates(body: Dict[str, Any]) -> requests.Response:
+
+        # Get wells from included data
+        wells = [elem for elem in self.included if elem["type"] == "wells"]
+
+        # For each well, get the control type for the well
+        # Create dict with key: well position, and value: control type
+        # control_types = { "A1": "pcr pos", "B1": None, .....}
+        control_types = {well["attributes"]["position"]["name"]: get_control_type_for_well(well) for well in wells}
+
+        # Filter out any wells which are not controls
+        return {k: v for (k, v) in control_types.items() if v is not None}
+
+
+def covert_json_response_into_dict(barcode: str, json) -> dict:
+    # Validate plate data exists
+    if len(json["data"]) == 0:
+        return {"data": None, "error": f"There is no plate data for barcode '{barcode}'"}
+
+    # Validate plate purpose is as expected
+    # plate_purpose = None
+    if (included := json["included"]) is not None:
+        purposes = [elem for elem in included if elem["type"] == "purposes"]
+
+        # Validate only one plate pupose exists
+        # TODO (DPL-572): check if is this needed? Could there ever be more than one purpose?
+        if len(purposes) != 1:
+            return {"data": None, "error": f"There should only be one purpose for barcode '{barcode}'"}
+
+        purpose_name = purposes[0]["attributes"]["name"]
+
+        if purpose_name != BIOSCAN_PLATE_PURPOSE:
+            return {"data": None, "error": f"Incorrect purpose '{purpose_name}' for barcode '{barcode}'"}
+
+        # Validate samples exists for plate
+        samples = [elem for elem in included if elem["type"] == "samples"]
+        if len(samples) == 0:
+            return {"data": None, "error": f"There are no samples for barcode '{barcode}'"}
+
+    # Instantiate wrapper class
+    control_locations = ControlLocations(included)
+
+    # Perform data processing, to retrieve well controls
+    control_info = control_locations.get_control_locations()
+    # { well_position: control_type } e.g.  {'A1': 'pcr positive', 'B1': 'pcr negative'}
+
+    # Get the well position, for the +ve and -ve control types
+    positive_control_position = [k for (k, v) in control_info.items() if v == "pcr positive"]
+    negative_control_position = [k for (k, v) in control_info.items() if v == "pcr negative"]
+
+    # Validate only one positive and one negative controls exist
+    if len(positive_control_position) > 1 or len(negative_control_position) > 1:
+        return {
+            "data": None,
+            "error": f"There should be only one positive and one negative control for barcode '{barcode}'",
+        }
+
+    # Validate both controls exist
+    if len(positive_control_position) != 1 or len(negative_control_position) != 1:
+        return {"data": None, "error": f"Missing positive or negative control for barcode '{barcode}'"}
+
+    # Convert data into expected format for Beckman
+    # This response was defined by the Beckman SAT
+    locations = {
+        "barcode": barcode,
```

Comment:
I wonder if the keys being used in these dictionaries should be constants too.

Reply:
Yes, they are constants now.